antiphishing/antiphishing.go

```diff
@@ -0,0 +1,183 @@
+	}
+	defer resp.Body.Close()
+	return domains, nil
+}
+
+func saveHyperfishDomains() ([]string, error) {
+	domains, err := fetchHyperfishDomains()
+	if err != nil {
+		return nil, err
+	}
+
+	//clear old domains incase there was a false positive
+	redisDelErr := common.RedisPool.Do(radix.Cmd(nil, "DEL", "hyperfish_domains"))
+	if redisDelErr != nil {
+		return nil, redisDelErr
+	}
+	//save new domains to set
+	args := append([]string{"hyperfish_domains"}, domains...)
+	redisSaveErr := common.RedisPool.Do(radix.Cmd(nil, "SADD", args...))
+	if redisSaveErr != nil {
+		return nil, redisSaveErr
+	}
+	return domains, nil
+}
+
+func queryHyperFish(input string) (*bool, error) {
+	isBadDomain := false
+	link, err := url.Parse(input)
+	if err != nil {
+		logrus.WithError(err).Error(`[antiphishing] failed to parse url`)
+		return nil, err
+	}
+	redisErr := common.RedisPool.Do(radix.FlatCmd(&isBadDomain, "SISMEMBER", "hyperfish_domains", link.Hostname()))
+	if redisErr != nil {
+		logrus.WithError(redisErr).Error(`[antiphishing] failed to check for hyperfish_domains`)
+		return nil, redisErr
+	}
+	return &isBadDomain, redisErr
+}
+
+func queryBitflowAntiFish(input []string) (*BitFlowAntiFishResponse, error) {
+	bitflowAntifishResponse := BitFlowAntiFishResponse{}
+	stringifiedUrlList := strings.Join(input, ",")
+	queryBytes, _ := json.Marshal(struct {
+		Message string `json:"message"`
+	}{stringifiedUrlList})
+	queryString := string(queryBytes)
+	client := &http.Client{}
+	req, _ := http.NewRequest("POST", bitflowAntiFishURL, strings.NewReader(queryString))
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Accept", "*/*")
+	req.Header.Add("Content-Length", strconv.Itoa(len(queryString)))
+	req.Header.Add("User-Agent", "YAGPDB")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		logrus.WithError(err).Error("[antiphishing] Failed checking bitflowAntiFish API ")
+		return nil, err
+	}
+
+	if resp.StatusCode == 404 {
+		bitflowAntifishResponse.Match = false
+		return &bitflowAntifishResponse, nil
+	}
+
+	if resp.StatusCode != 200 {
+		respError := fmt.Errorf("[antiphishing] Unable to fetch data from AntiFish API, status-code %d", resp.StatusCode)
+		logrus.WithError(respError)
+		return nil, respError
+	}
+
+	req.Body.Close()
+	defer resp.Body.Close()
+
+	bytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		logrus.WithError(err).Error("[antiphishing] Error parsing response body from bitflowAntiFish API")
+		return nil, err
+	}
+
+	jsonErr := json.Unmarshal(bytes, &bitflowAntifishResponse)
+	if err != nil {
+		logrus.WithError(jsonErr).Error(("[antiphishing] Error parsing JSON from bitflowAntiFish API"))
+		return nil, jsonErr
+	}
+
+	return &bitflowAntifishResponse, nil
+}
+
+func checkPhishingDomains(input []string) (*string, error) {
+	i := 0
+	for range input {
+		isPhishingLink, err := queryHyperFish(input[i])
+		if err != nil {
+			return nil, err
+		}
+		if *isPhishingLink {
+			return &input[i], err
+		}
+	}
```

Comment:
```suggestion
	for i, d := input {
		isPhishingLink, err := queryHyperFish(d)
		if err != nil {
			return nil, err
		}
		if *isPhishingLink {
			return &d, err
		}
	}
```
(More idiomatic. Also, presently the code appears incorrect as `i` is always `0` as opposed to iterating through `input`.)